pkg/util/git.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package util
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// LoadRepository loads a repository from the local file system.
+// TODO(a-hilaly): load repository into a memory filesystem (needs go1.16
```

Comment:
How large would this be in memory? I know the `aws-sdk-go` has a large commit history and many, many tags.

Reply:
```shell
du -sh ~/.cache/aws-controllers-k8s/src/aws-sdk-go 
352M	/Users/hilalymh/.cache/aws-controllers-k8s/src/aws-sdk-go
```

I guess it's gonna be ~350M - I will do some profile on `ack-generate` binary before implementing it

Reply:
@RedbackThomson Is it a bad a idea to load such a big repository in memory?